mktrayicon.c

```diff
@@ -256,10 +300,99 @@ gpointer watch_fifo(gpointer argv)
 #endif
 			break;
 		case 'm': /* menu */
-			fprintf(stderr, "Menu command not yet implemented\n");
-			if (param != NULL) {
-				free(param);
+			if (onmenu != NULL){
+				//destroy the previous menu
+				for (int i = 0; i < menusize; i++){
+					free(onmenu[i].name);
+					free(onmenu[i].action);
+					onmenu[i].name = NULL;
+					onmenu[i].action = NULL;
+				}
+				free(onmenu);
+				onmenu = NULL;
+				gtk_widget_destroy(menu);
+				menu = NULL;
 			}
+			
+			menusize = 0;
+
+			if (param != NULL && *param == '\0') {
+#ifdef DEBUG
+				printf("Removing onmenu handler\n");
+#endif
+				break;
+			}
+
+			// This block makes sure that the parameter after 'm' is well defined
+			int commas = 0;
+			int bars = 0;
+			if (len < 3){
+				break;
```

Comment:
I think this and all the other `break`s below have to `free(param)`

Reply:
You are right. I forgot to check this. Since we don't use it anymore we have to free.